packages/dara-core/dara/core/internal/websocket.py

```diff
@@ -168,11 +172,23 @@ async def process_client_message(self, message: ClientMessage):
             # If the message has a channel ID, it's a response to a previous message
             if message_id:
                 if message_id in self.pending_responses:
-                    event, _ = self.pending_responses[message_id]
+                    event, existing_messages = self.pending_responses[message_id]
+
+                    # If the response is chunked then collect the messages in pending responses
+                    if message.chunk_count is not None:
+                        if existing_messages is not None and isinstance(existing_messages, list):
+                            existing_messages.append(message.message)
+                        else:
+                            existing_messages = [message.message]
+                            self.pending_responses[message_id] = (event, existing_messages)
 
-                    # Store the response and set the event to notify the waiting coroutine
-                    self.pending_responses[message_id] = (event, message.message)
-                    event.set()
+                        # If all chunks have been received, set the event to notify the waiting coroutine
+                        if len(existing_messages) == message.chunk_count:
+                            event.set()
+                    else:
+                        # Store the response and set the event to notify the waiting coroutine
+                        self.pending_responses[message_id] = (event, message.message)
+                        event.set()
 
         if message.type == 'custom':
```

Comment:
Do we want the same chunking logic for custom messages, i.e. wait for all chunks and combine them before passing into user handler? Otherwise it will only work for internal messages

Reply:
Happy to add for customer messages as well, just wanted to verify the approach before making too many changes.